content/posts/tekton-jx-pipelines.md

```diff
@@ -0,0 +1,954 @@
+spec:
+  resources:
+    - name: source-repo
+      type: git
+    - name: docker-container
+      type: image
+    - name: deploy-repo
+      type: git
+  tasks:
+    - name: petclinic-maven
+      taskRef:
+        name: build-maven
+      resources:
+        inputs:
+          - name: workspace
+            resource: source-repo
+        outputs:
+          - name: workspace
+            resource: source-repo
+    - name: petclinic-kaniko
+      taskRef:
+        name: build-kaniko
+      resources:
+        inputs:
+          - name: workspace
+            resource: source-repo
+            from: 
+              - petclinic-maven
+        outputs:
+          - name: dockerImage
+            resource: docker-container
+    - name: petclinic-deploy
+      taskRef:
+        name: deploy-kubectl
+      runAfter:
+        - petclinic-kaniko
+      resources:
+        inputs:
+          - name: workspacedeploy
+            resource: deploy-repo
+      params:
+        - name: deployFile
+          value: test-deploy-secret.yaml
+```
+
+Then we can define also a file `pipeline-resources.yaml` with all required `PipelineResources` objects needed:
+```yaml
+apiVersion: tekton.dev/v1alpha1
+kind: PipelineResource
+metadata:
+  name: petclinic-git
+spec:
+  type: git
+  params:
+    - name: url
+      value: https://github.com/dcanadillas/petclinic-kaniko.git
+---
+apiVersion: tekton.dev/v1alpha1
+kind: PipelineResource
+metadata:
+  name: petclinic-image
+spec:
+  type: image
+  params:
+    - name: url
+      value: eu.gcr.io/emea-sa-demo/petclinic-kaniko:latest
+---
+apiVersion: tekton.dev/v1alpha1
+kind: PipelineResource
+metadata:
+  name: petclinic-deploy
+spec:
+  type: git
+  params:
+    - name: url
+      value: https://github.com/dcanadillas/petclinic-kaniko-deploy.git
+```
+
+As `PipelineResources` we are using the two GitHub repositories from the original repos as inputs (one for the application and the other for deployment definitions), and one container image as output to be built.
+
+Next, let's define the `PipelineRun`, that executes and instantiate the pipeline, in a file `petclinic-run.yaml`:
+
+```yaml
+apiVersion: tekton.dev/v1alpha1
+kind: PipelineRun
+metadata:
+  name: petclinic-pipelinerun
+spec:
+  pipelineRef:
+    name: petclinic-pipeline
+  serviceAccount: 'default'
+  serviceAccounts:
+    - taskName: petclinic-deploy
+      serviceAccount: tekton-deployment
+  resources:
+  - name: source-repo
+    resourceRef:
+      name: petclinic-git
+  - name: docker-container
+    resourceRef:
+      name: petclinic-image
+  - name: deploy-repo
+    resourceRef:
+      name: petclinic-deploy
+```
+
+In the `PipelineRun` object is important to understand that is intended in the way of defining the specific running instance of a deployed pipeline. It's the right place to assign running input/output parameter values, specify `serviceAccounts` to be used, or referencing the right `PipelieResources` to be used.
```

Comment:
> In the `PipelineRun` object is important to understand that is intended in the way

Missing a couple of 'it's in there, should be:

In the `PipelineRun` object it is important to understand that it is intended in the way